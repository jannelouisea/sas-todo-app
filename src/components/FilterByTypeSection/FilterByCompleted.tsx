import {
    FilterByButton
} from 'src/components'
import DoneIcon from '@mui/icons-material/Done';
import PropTypes from 'prop-types';
import {
    MUIColor
} from 'src/static/enums'

import { useAppDispatch, useAppSelector } from 'src/app/hooks';
import { showCompletedToggled } from 'src/features/todoItemsFilters/todoItemsFiltersSlice';

type Props = {
    className: string
}

function FilterByCompleted({ className }: Props) {
    const showCompleted = useAppSelector(state => state.todoItemsFilters.showCompleted);
    const dispatch = useAppDispatch();

    const onToggleFilter = () => {
        dispatch(showCompletedToggled());
    }

    return <FilterByButton
        className={className}
        label='Completed'
        color={MUIColor.Success}
        startIcon={<DoneIcon fontSize='small' />}
        filterEnabled={showCompleted}
        onClick={onToggleFilter}
    />;
}

FilterByCompleted.propTypes = {
    className: PropTypes.string,
}

FilterByCompleted.defaultProps = {
    className: ''
}

export default FilterByCompleted;