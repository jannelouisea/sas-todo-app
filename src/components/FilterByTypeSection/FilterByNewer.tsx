import {
    FilterByButton
} from 'src/components'
import HourglassTopIcon from '@mui/icons-material/HourglassTop';
import PropTypes from 'prop-types';

import { useAppDispatch, useAppSelector } from 'src/app/hooks';
import { showNewerToggled } from 'src/features/todoItemsFilters/todoItemsFiltersSlice';

type Props = {
    className: string
}

function FilterByNewer({ className }: Props) {
    const showNewer = useAppSelector(state => state.todoItemsFilters.showNewer);
    const dispatch = useAppDispatch();

    const onToggleFilter = () => {
        dispatch(showNewerToggled());
    }

    return <FilterByButton
        className={className}
        label='Newer'
        color='primary'
        startIcon={<HourglassTopIcon fontSize='small' />}
        filterEnabled={showNewer}
        onClick={onToggleFilter}
    />;
}

FilterByNewer.propTypes = {
    className: PropTypes.string,
}

FilterByNewer.defaultProps = {
    className: ''
}

export default FilterByNewer;