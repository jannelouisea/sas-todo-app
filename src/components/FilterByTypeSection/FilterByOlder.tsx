import {
    FilterByButton
} from 'src/components'
import HourglassBottomIcon from '@mui/icons-material/HourglassTop';

import { useAppDispatch, useAppSelector } from 'src/app/hooks';
import { showOlderToggled } from 'src/features/todoItemsFilters/todoItemsFiltersSlice';
import PropTypes from 'prop-types';

type Props = {
    className: string
}

function FilterByOlder({ className }: Props) {
    const showOlder = useAppSelector(state => state.todoItemsFilters.showOlder);
    const dispatch = useAppDispatch();

    const onToggleFilter = () => {
        dispatch(showOlderToggled());
    }

    return <FilterByButton
        className={className}
        label='Older'
        color='warning'
        startIcon={<HourglassBottomIcon fontSize='small' />}
        filterEnabled={showOlder}
        onClick={onToggleFilter}
    />;
}

FilterByOlder.propTypes = {
    className: PropTypes.string,
}

FilterByOlder.defaultProps = {
    className: ''
}

export default FilterByOlder;