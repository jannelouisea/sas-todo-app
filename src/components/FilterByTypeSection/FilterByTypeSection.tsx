import {
    FilterByNewer,
    FilterByOlder,
    FilterByCompleted,
} from 'src/components';

function FilterByTypeSection() {
    return (
        <div className='flex'>
            <div className='mr-2'><FilterByNewer /></div>
            <div className='mr-2'><FilterByOlder /></div>
            <FilterByCompleted />
        </div>
    );
}

export default FilterByTypeSection;