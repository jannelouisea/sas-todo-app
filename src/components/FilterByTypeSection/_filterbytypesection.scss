.filter-by-type-section {
    .filter-types {
        display: flex;

        .filter-by-type {
            margin-right: .25rem;
        }
    }
}