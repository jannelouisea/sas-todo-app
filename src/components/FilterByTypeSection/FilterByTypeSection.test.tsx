import renderer from 'react-test-renderer';
import FilterByTypeSection from './FilterByTypeSection';
import { Provider } from 'react-redux';
import {
    IStoreType,
    createStore
} from 'src/app/store';

describe('FilterByTypeSection', () => {
    let store: IStoreType;

    beforeEach(() => {
        store = createStore();
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <FilterByTypeSection className='foo' />
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});