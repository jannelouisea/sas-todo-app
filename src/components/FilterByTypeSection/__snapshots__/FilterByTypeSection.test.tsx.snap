// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByTypeSection renders correctly 1`] = `
<div
  className="filter-by-type-section foo"
>
  <div
    className=""
  >
    <p
      className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-1v5xtsw-JoyTypography-root"
    >
      Filter by
    </p>
  </div>
  <div
    className="filter-types"
  >
    <div
      className="filter-by-type"
    >
      <button
        aria-label="Filter by newer"
        className="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeSm filter-by-type css-1d374uw-JoyButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="HourglassTopIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m6 2 .01 6L10 12l-3.99 4.01L6 22h12v-6l-4-4 4-3.99V2H6zm10 14.5V20H8v-3.5l4-4 4 4z"
            />
          </svg>
        </span>
        Newer
      </button>
    </div>
    <div
      className="filter-by-type"
    >
      <button
        aria-label="Filter by older"
        className="MuiButton-root MuiButton-variantSolid MuiButton-colorWarning MuiButton-sizeSm filter-by-type css-16bqlfy-JoyButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="HourglassBottomIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
            />
          </svg>
        </span>
        Older
      </button>
    </div>
    <div
      className="filter-by-type"
    >
      <button
        aria-label="Filter by completed"
        className="MuiButton-root MuiButton-variantSolid MuiButton-colorSuccess MuiButton-sizeSm filter-by-type css-1yqcigs-JoyButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="DoneIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
            />
          </svg>
        </span>
        Completed
      </button>
    </div>
  </div>
</div>
`;
